lib/dust.js

```diff
@@ -565,9 +585,36 @@
     return this;
   };
 
+  /**
+   * Pipes to a WritableStream. Note that backpressure isn't implemented,
+   * so we just write as fast as we can.
+   * @param stream {WritableStream}
+   * @return self
+   */
   Stream.prototype.pipe = function(stream) {
+    if(typeof stream.write !== 'function' ||
+       typeof stream.end !== 'function') {
+      dust.log('Incompatible stream passed to `pipe`', WARN);
+      return this;
+    }
+
+    var destEnded = false;
+
+    if(typeof stream.emit === 'function') {
```

Comment:
Just Dust convention TBH, I don't mind what we use.

But most of them are already nouns like `reference`, `section`, `block`. Steven asked for something other than `thenable` so we went with `await`.

Stream is a verb and noun so it makes sense either way? :D